nixos/modules/services/databases/postgresql.nix

```
@@ -462,7 +498,12 @@ in
   config = mkIf cfg.enable {
 
     assertions = map ({ name, ensureDBOwnership, ... }: {
-      assertion = ensureDBOwnership -> elem name cfg.ensureDatabases;
+      assertion = ensureDBOwnership -> any (db:
+        if builtins.isString db then
+          db == name
+        else
+          db.name == name
+      ) cfg.ensureDatabases;
```

Would then mean you don't need this `if`:

```suggestion
      assertion = ensureDBOwnership -> any (db: db.name == name) cfg.ensureDatabases;
```